### PR TITLE
9824: fix undefined variables and pylint errors

### PR DESCRIPTION
**Pull request by romjerome**

### CODE REVIEW

File: gramps/plugins/db/bsddb/bsddb.gpr.py [hunk 1]
```diff
@@ -17,6 +17,9 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
+from gramps.gen.const import GRAMPS_LOCALE as glocale
+_ = glocale.translation.gettext
```
Comment (Nick-Hall) [inline gramps/plugins/db/bsddb/bsddb.gpr.py]: This line should come after the imports.  I normally move it together with its import.

File: gramps/plugins/db/bsddb/summary.py [hunk 2]
```diff
@@ -26,6 +26,7 @@
 
 from gramps.gen.db import META, PERSON_TBL
 from  gramps.gen.db.dbconst import BDBVERSFN
+from gramps.gen.db.generic import LOG
```
Comment (Nick-Hall) [inline gramps/plugins/db/bsddb/summary.py]: I don't think you want to import `LOG` from here.

File: gramps/plugins/db/bsddb/write.py [hunk 3]
```diff
@@ -72,6 +72,7 @@
 
 from gramps.gen.db import exceptions
```
Comment (Nick-Hall) [inline gramps/plugins/db/bsddb/write.py]: Is this import still needed?

File: gramps/plugins/importer/importprogen.py [hunk 4]
```diff
@@ -1031,6 +1031,7 @@ def create_persons(self):
                     loc.set_postal_code(postal)
                     loc.set_country(country)
                     loc.set_phone(telno)
+                    # TO_FIX
```
Comment (Nick-Hall) [inline gramps/plugins/importer/importprogen.py]: It is convention to use FIXME rather than TO_FIX.

File: gramps/plugins/db/bsddb/summary.py (hunk 2 above)
Comment (romjerome) [inline gramps/plugins/db/bsddb/summary.py]: Oh, yes my bad!

```
import logging
LOG = logging.getLogger(".dbsummary")
```

Thanks!

File: gramps/plugins/db/bsddb/write.py (hunk 3 above)
Comment (romjerome) [inline gramps/plugins/db/bsddb/write.py]: Rather a consistency issue?

> $ git grep "exceptions" gramps/plugins/db/bsddb/write.py
> gramps/plugins/db/bsddb/write.py:from gramps.gen.db import exceptions
> gramps/plugins/db/bsddb/write.py:        one of the exceptions in DBERRS, the error is logged and a DbError
> gramps/plugins/db/bsddb/write.py:                raise exceptions.BsddbUpgradeRequiredError(bsddb_version,
> gramps/plugins/db/bsddb/write.py:                raise exceptions.BsddbDowngradeRequiredError(bsddb_version,
> gramps/plugins/db/bsddb/write.py:            raise exceptions.PythonDowngradeError(db_python_version,
> gramps/plugins/db/bsddb/write.py:                raise exceptions.PythonUpgradeRequiredError(db_python_version,
> gramps/plugins/db/bsddb/write.py:        The purpose of this function is to catch any exceptions that occur.
> gramps/plugins/db/bsddb/write.py:            raise exceptions.DbException(str(msg))
> gramps/plugins/db/bsddb/write.py:        The purpose of this function is to catch any exceptions that occur.

**Comment by romjerome** [comment]
Maybe I should try to complete all plugins (more plugins) before a Pull Request?

File: gramps/plugins/db/bsddb/bsddb.gpr.py (hunk 1 above)
Comment (romjerome) [inline gramps/plugins/db/bsddb/bsddb.gpr.py]: Oh, yes. 

I am also trying to limit [coverage diff](https://github.com/codecov/support/wiki/Coverage-Diff), but sometimes with [unexpected returns](https://github.com/codecov/support/wiki/Unexpected-Coverage-Changes).

**Comment by romjerome** [comment]
Simple changes have been included on master.

Remaining some experimental works around:
 some pylint errors around CLI 			cd39217
 try to fix Serbian Date handler (pylint)    2e72a8c
 try to fix errors on gen modules 		2c2199f
 try to fix errors on gui modules                1be7e12

**Comment by Nick-Hall** [comment]
It looks like this needs a rebase.

**Comment by romjerome** [comment]
> It looks like this needs a rebase.

Yes, it looks like ... 
Its seems that all commits from "origin/romjerome-master" are now on "origin/master"!

The best is maybe to close this pull request?
Annoying issues have been already fixed on master branch.

Some variables are still missing (for pylint)
but I [did not push the change](https://github.com/gramps-project/gramps/pull/284#issuecomment-268510684) because
I was not always certain to understand the way used
by the related code. Also, variable issues around error module or LOG,
could have a low priority as it raises an error, anyway. It did not raise
since a while, so they are rather typos.

I will use a safe way: new branch with current gramps/master code,
merge my origin/master with this new branch, create a new branch 
from my origin/master, patch files, create one pull request.

**Comment by tosky** [comment]
> I will use a safe way: new branch with current gramps/master code,
merge my origin/master with this new branch, create a new branch
from my origin/master, patch files, create one pull request.

Please rebase your origin/master on top of the new branch, not merge it. This way the changes will be limited to the real code changes and not other cruft (merge commits for example).

**Comment by romjerome** [comment]
@tosky 

I tried to properly rebase, but got too many conflicts...

Now, rather use the trick pointed out by @jralls :
`git checkout -b foo --no-track gramps/master `
than generate one pull request, like #309 

Note, there is a possible change against a [file used by the Italian locale](https://github.com/gramps-project/gramps/pull/284/files#diff-bdb20681c0eafdb298a03ec7f75eb6bb)!

I will not included it into a pull request. If need, it is raised by such type of command:
`pylint -E  gramps/plugins/rel/rel_it.py` /see bug report 9824/

'pylint' can be wrong and sometimes the raised error is cosmetic or a false/positive,
but most part of time the error exists. Here, we know that the migration is not done.
As there is a cultural issue around Italian, I do not know if the fix is right (does it break something?) or if we really need it. Maybe just need to rewrite and review this section.

Thanks.

**Comment by romjerome** [comment]
Everything should be now splitten either on new branches (under fork) or pull requests.

The remaining issues for #284 are listed on [this branch](https://github.com/romjerome/gramps_cryptic_rebase/compare/remain). Except for "rel_fr.py", which could 
be already pushed on master gramps, the remaining proposal are often experimental...

Pylint does not raise errors for them, but I am not certain of the quality of the solutions used
for fixing them or how to test these errors pointed out by pylint!

Note, some errors are maybe still present because only a small part of modules has been tested on bug report 9824.

**Comment by romjerome** [comment]
@jralls 
Conflicts on pull requests have been fixed and pushed on romjerome:master
but problems remain, rebase fails and gitk does not help as your commits (PR)
are not visible on my local copy. Pull returns nothing (up-to-date)!

I suppose there is a remote mistake somewhere as romjerome:romjerome-master
is always ahead and pushing from my master branch does nothing.

This Pull Request seems to also bloc command like rebase on romjerome:master!

Sorry.

**Comment by romjerome** [comment]
```
$ git show fb5a70f
commit fb5a70fce1f03e4776f21632135b5aedc6091699
Merge: c7357fd 6583e2e
Author: romjerome <romjerome@users.noreply.github.com>
Date:   Wed Jan 4 15:05:06 2017 +0100
    Merge pull request #5 from jralls/bug9595
    Bug9595
```
This commit is hidden on a command like git log!
This is really confusing. You wrote that we should ignore
commits in such case, but blind work even via GUI, is
maybe something that I cannot do at a glance; need
to trust our git commands... 

Maybe we should close this pull request?
The main changes are already on gramps:master.

I will commit rel_fr.py, later.

Thank you.

**Comment by romjerome** [comment]
> I will commit rel_fr.py, later.

[done](https://github.com/gramps-project/gramps/commit/2a1234c45c049fa77ed0b364afc205ef4eed85aa).

The remaining experimental fixes for variables (pylint) on gramps/master, should be rather close to [this](https://github.com/romjerome/gramps_cryptic_rebase/compare/remain).
this.

**Comment by jralls** [comment]
@romjerome I suggest that you delete both branches master and romjerome-master from your repository. They have too many cross-merges in them and so are really messy. You won't ever be able to cleanly merge anything to or from them.
 
Create a new local master branch that tracks gramps-project/gramps/master and rebase my PR branches on to that.

**Comment by romjerome** [comment]
Agreed, but this could be a little bit dangerous, now, as I am testing some local modifications
around [place separator](https://github.com/romjerome/gramps_cryptic_rebase/tree/place_separators), where I just see that [place details gramplet could also have a local change](https://github.com/gramps-project/gramps/blob/master/gramps/plugins/gramplet/placedetails.py#L154)...

Also, maybe I should rather try to use a [GEP041 branch](https://github.com/romjerome/gramps_cryptic_rebase/tree/gep-041-new-selector) on gramps repository?

Currently, only using fork as a github server copy...

Does removing HEAD stuff (or master branch) on my fork will not break these branches too?

**Comment by jralls** [comment]
Move your tests to a new branch. Now. Before you get tempted to test something else on that branch.

I can't think of a good reason to make a GEP041 branch on gramps-project. You can use your own github repo to publish it for reviews or collaboration. Do make a branch and do rebase it often on gramps-project/gramps/master, then force-push the rebases to romjerome/gramps/GEP041.

Think of your fork as your public sandbox and remote backup. You can use it to share your work-in-progress. 

No, branch HEADs are just pointers to commits. Git isn't like Subversion where a branch is an actual physical copy of changes. A Git repo is a tree of back-linked commits (where each commit includes a reference to one or more parents, where branches and tags are just references to a commit. Deleting a branch removes the reference, but the commit is still there until ```git gc``` runs (every 100 commits by default) and cleans up all of the commits that don't have something pointing at them. Read the [tripes de Git](https://git-scm.com/book/fr/v2/Les-tripes-de-Git-Plomberie-et-porcelaine) chapter in Chacon's book for the gory details. It won't take long and it's well worth the time.

**Comment by romjerome** [comment]
@jralls 
yes, it seems that I was cooking spaghetti...
```
$ git branch -d master
warning: suppression de la branche 'master' qui a été fusionnée dans
         'refs/remotes/origin/romjerome-master', mais pas dans HEAD.
Branche master supprimée (précédemment 6d39031).

$ git show 6d39031
commit 6d390314022bf6d4f3738e8d8f110eb700f959d8
Merge: b131fd9 8a58115
Date:   Fri Jan 6 11:17:44 2017 +0100

    Merge branch 'gep-041-new-selector' into romjerome-master
```
Now, will create a new master branch ('romjerome-master' was only an other copy)

**Comment by romjerome** [comment]
```
$ git checkout -b master gramps/master
La branche master est paramétrée pour suivre la branche distante master depuis gramps.
Basculement sur la nouvelle branche 'master'
```
> rebase my PR branches on to that
I need to find the uri/url/path for your PR branches as a remote one?

**Comment by jralls** [comment]
Your fixed-up branches are all on [my gramps repo](https://github.com/jralls/gramps.git). The easiest way to integrate them at this point would be to add that as a remote on your *local* repo and fetch it:
```
git remote add jralls https://github.com/jralls/gramps.git
git fetch jralls
```
Then you'll have them all in your repo and you can refer to them as e.g. ```jralls/bug5742```. For example, to create your own bug5742 branch and get it ready to merge into master:
```
git checkout -b bug5742 998c5f4 #That's the commit where jralls/bug5742 connects to master
git merge --ff-only jralls/bug5742 #And starting your bug5742 there is what allows the fast-forward merge
git rebase master bug5742 #This recreates the branch rooted at the current master.
```
Once you've resolved any new conflicts and tested thoroughly to make sure that nothing got messed up in all of the rebasing (both mine and yours) you can merge it back to master. Since you named every commit with the bug # there's no need to have a merge commit so you can let it fast forward.
```
git checkout master
git merge bug5742
git push gramps-project master # Use the real remote name that points to gramps-project, of course.
```
Repeat for each bug-fix branch in turn. You'll do the first part for the geps041 branch, but I suppose that's not ready to push to master, so you'll want to keep it as a separate branch until it is.

Once you're done you can lose my repo and recover the disk space:
```
git remote rm jralls
```

**Comment by romjerome** [comment]
Sorry, but after a rebase and a push, these commits have been included on gramps/master and this PR has been closed !!! That's annoying, might be related to a commit ahead (fix pylint warnings on 'const.py'). :-(

During rebase, the return printed something about 3 points (commits?).
Should I revert them? Anyway, I will remove this master branch once more
as this was not expected.

**Comment by romjerome** [comment]
On the other hand, maybe I should try to run last revisions for pylint?
The last code includes a [new check](https://github.com/PyCQA/pylint/commit/4bb474dfad2d2dd8ea357f6b8e6a1c708246ac4a) and a [fix](https://github.com/PyCQA/pylint/commit/da1da56853380a5a387ad287f4398402b14ef123). Both could be in relation with this PR?

**Comment by jralls** [comment]
The PR closed because you replaced your master branch with a clean one. If you want to use PRs for the new branches you need to create them, one for each branch.

I don't recognize the provenance of the two commits you pushed, but Travis didn't like them so you should either revert or push a fix.

**Comment by romjerome** [comment]
@Nick-Hall reverted my mistake.
Thank you.

Note, the problem remains, should I revert the commit for this PR?
Theses recent modified files were not hardly tested, it was rather a local fix.

**Comment by romjerome** [comment]
Really strange, the commit for this PR has been reverted!

> romjerome:master was force-pushed and no longer has any new commits.
> Pushing new commits will allow the pull request to be re-opened.

Well, I will remove romjerome:master ... and will no more create a pull request from
romjerome:master.

Sorry.

**Comment by jralls** [comment]
Nick reverted the "fix warnings" commit, which fixed Travis temporarily, and now there are new commits rebased on your changes from 8a8807c, so I suggest that you get current with master and test on that. Push any fixes that you need to make in a new commit.

**Pull request closed without merging**